### ISSUE TITLE
Unable to locate a class or view for component [splade-component] after upgrade to 1.2.12

### ISSUE THREAD

**Issue by jamesj2** [protonemedia/laravel-splade#303]
- Laravel Version: 9.51.0
- PHP Version: 8.1.14
- Splade JS Version (npm): 1.2.12
- Splade PHP Version (composer): 1.2.14

### Description:

I working on refactoring a large legacy app.  And I upgraded from version 1.2.11.1 to 1.2.12 to try out the new global data feature.  Now my pages won't load.  I'm getting `Unable to locate a class or view for component [splade-component]`.  I tried removing my customized splade views and config files.  No luck.  The example app is working fine but I'm unable to determine where the fault is.  If I downgrade to 1.2.11.1 it fixes my issue.

### Steps To Reproduce Issue:

```
composer update protonemedia/laravel-splade
npm upgrade @protonemedia/laravel-splade
php artisan view:clear
```

### Stacktrace:

```
[2023-02-10 14:03:51] local.ERROR: InvalidArgumentException: Unable to locate a class or view for component [splade-component]. in app/vendor/laravel/framework/src/Illuminate/View/Compilers/ComponentTagCompiler.php:311
145   │ Stack trace:
146   │ #0 app/vendor/laravel/framework/src/Illuminate/View/Compilers/ComponentTagCompiler.php(235): Illuminate\View\Compilers\ComponentTagCompiler->componentClass('splade-componen...')
147   │ #1 app/vendor/laravel/framework/src/Illuminate/View/Compilers/ComponentTagCompiler.php(220): Illuminate\View\Compilers\ComponentTagCompiler->componentString('splade-componen...', Array)
148   │ #2 [internal function]: Illuminate\View\Compilers\ComponentTagCompiler->Illuminate\View\Compilers\{closure}(Array)
149   │ #3 app/vendor/laravel/framework/src/Illuminate/View/Compilers/ComponentTagCompiler.php(215): preg_replace_callback('/\n            <...', Object(Closure), '<x-splade-compo...')
150   │ #4 app/vendor/laravel/framework/src/Illuminate/View/Compilers/ComponentTagCompiler.php(89): Illuminate\View\Compilers\ComponentTagCompiler->compileSelfClosingTags('<x-splade-compo...')
151   │ #5 app/vendor/laravel/framework/src/Illuminate/View/Compilers/ComponentTagCompiler.php(76): Illuminate\View\Compilers\ComponentTagCompiler->compileTags('<x-splade-compo...')
152   │ #6 app/vendor/laravel/framework/src/Illuminate/View/Compilers/BladeCompiler.php(423): Illuminate\View\Compilers\ComponentTagCompiler->compile('<x-splade-compo...')
153   │ #7 app/vendor/protonemedia/laravel-splade/src/CustomBladeCompiler.php(24): Illuminate\View\Compilers\BladeCompiler->compileComponentTags('<x-splade-compo...')
154   │ #8 app/vendor/laravel/framework/src/Illuminate/View/Compilers/BladeCompiler.php(256): ProtoneMedia\Splade\CustomBladeCompiler->compileComponentTags('<x-splade-compo...')
155   │ #9 app/vendor/laravel/framework/src/Illuminate/View/Compilers/BladeCompiler.php(175): Illuminate\View\Compilers\BladeCompiler->compileString('<x-splade-compo...')
156   │ #10 app/vendor/laravel/framework/src/Illuminate/View/Engines/CompilerEngine.php(62): Illuminate\View\Compilers\BladeCompiler->compile('...')
157   │ #11 app/vendor/laravel/framework/src/Illuminate/View/View.php(195): Illuminate\View\Engines\CompilerEngine->get('...', Array)
158   │ #12 app/vendor/laravel/framework/src/Illuminate/View/View.php(178): Illuminate\View\View->getContents()
159   │ #13 app/vendor/laravel/framework/src/Illuminate/View/View.php(147): Illuminate\View\View->renderContents()
160   │ #14 app/vendor/laravel/framework/src/Illuminate/View/Compilers/BladeCompiler.php(320): Illuminate\View\View->render()
161   │ #15 app/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(338): Illuminate\View\Compilers\BladeCompiler::render('<x-splade-compo...')
162   │ #16 app/vendor/protonemedia/laravel-splade/src/Http/SpladeMiddleware.php(478): Illuminate\Support\Facades\Facade::__callStatic('render', Array)
163   │ #17 app/vendor/protonemedia/laravel-splade/src/Http/SpladeMiddleware.php(232): ProtoneMedia\Splade\Http\SpladeMiddleware::renderedComponents()
164   │ #18 app/vendor/protonemedia/laravel-splade/src/Http/SpladeMiddleware.php(93): ProtoneMedia\Splade\Http\SpladeMiddleware->handleRegularRequest(Object(Illuminate\Http\Request), Object(Illuminate\Http\Response), Object(stdClass))
165   │ #19 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): ProtoneMedia\Splade\Http\SpladeMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
166   │ #20 app/app/Http/Middleware/Navigation.php(25): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
167   │ #21 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): App\Http\Middleware\Navigation->handle(Object(Illuminate\Http\Request), Object(Closure))
168   │ #22 app/app/Http/Middleware/UserActivity.php(34): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
169   │ #23 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): App\Http\Middleware\UserActivity->handle(Object(Illuminate\Http\Request), Object(Closure))
170   │ #24 app/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(50): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
171   │ #25 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Routing\Middleware\SubstituteBindings->handle(Object(Illuminate\Http\Request), Object(Closure))
172   │ #26 app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
173   │ #27 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle(Object(Illuminate\Http\Request), Object(Closure))
174   │ #28 app/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
175   │ #29 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure))
176   │ #30 app/app/Http/Middleware/HashTag.php(28): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
177   │ #31 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): App\Http\Middleware\HashTag->handle(Object(Illuminate\Http\Request), Object(Closure))
178   │ #32 app/app/Http/Middleware/PasswordAttempts.php(28): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
179   │ #33 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): App\Http\Middleware\PasswordAttempts->handle(Object(Illuminate\Http\Request), Object(Closure))
180   │ #34 app/app/Http/Middleware/CheckStatus.php(19): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
181   │ #35 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): App\Http\Middleware\CheckStatus->handle(Object(Illuminate\Http\Request), Object(Closure))
182   │ #36 app/vendor/spatie/laravel-http-logger/src/Middlewares/HttpLogger.php(28): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
183   │ #37 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Spatie\HttpLogger\Middlewares\HttpLogger->handle(Object(Illuminate\Http\Request), Object(Closure))
184   │ #38 app/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
185   │ #39 app/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\Session\Middleware\StartSession->handleStatefulRequest(Object(Illuminate\Http\Request), Object(Illuminate\Session\Store), Object(Closu
│ re))
186   │ #40 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
187   │ #41 app/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
188   │ #42 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
189   │ #43 app/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
190   │ #44 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
191   │ #45 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
192   │ #46 app/vendor/laravel/framework/src/Illuminate/Routing/Router.php(799): Illuminate\Pipeline\Pipeline->then(Object(Closure))
193   │ #47 app/vendor/laravel/framework/src/Illuminate/Routing/Router.php(776): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
194   │ #48 app/vendor/laravel/framework/src/Illuminate/Routing/Router.php(740): Illuminate\Routing\Router->runRoute(Object(Illuminate\Http\Request), Object(Illuminate\Routing\Route))
195   │ #49 app/vendor/laravel/framework/src/Illuminate/Routing/Router.php(729): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
196   │ #50 app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(190): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
197   │ #51 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
198   │ #52 app/vendor/raveren/laravel-debugbar-phpstorm/src/Middleware/InjectDebugbarVscode.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
199   │ #53 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): ErlangParasu\DebugbarVscode\Middleware\InjectDebugbarVscode->handle(Object(Illuminate\Http\Request), Object(Closure))
200   │ #54 app/vendor/barryvdh/laravel-debugbar/src/Middleware/InjectDebugbar.php(59): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
201   │ #55 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Barryvdh\Debugbar\Middleware\InjectDebugbar->handle(Object(Illuminate\Http\Request), Object(Closure))
202   │ #56 app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
203   │ #57 app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
204   │ #58 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull->handle(Object(Illuminate\Http\Request), Object(Closure))
205   │ #59 app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
206   │ #60 app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
207   │ #61 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Foundation\Http\Middleware\TrimStrings->handle(Object(Illuminate\Http\Request), Object(Closure))
208   │ #62 app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
209   │ #63 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle(Object(Illuminate\Http\Request), Object(Closure))
210   │ #64 app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
211   │ #65 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance->handle(Object(Illuminate\Http\Request), Object(Closure))
212   │ #66 app/vendor/laravel/framework/src/Illuminate/Http/Middleware/HandleCors.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
213   │ #67 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Http\Middleware\HandleCors->handle(Object(Illuminate\Http\Request), Object(Closure))
214   │ #68 app/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
215   │ #69 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Http\Middleware\TrustProxies->handle(Object(Illuminate\Http\Request), Object(Closure))
216   │ #70 app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
217   │ #71 app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(165): Illuminate\Pipeline\Pipeline->then(Object(Closure))
218   │ #72 app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(134): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
219   │ #73 app/public/index.php(51): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
220   │ #74 /Users/james/.composer/vendor/laravel/valet/server.php(234): require('...')
221   │ #75 {main}  {"file":"app/app/Exceptions/Handler.php","line":66,"class":"App\\Exceptions\\Handler","callType":"->","function":"report"}

```

**Comment by pascalbaljet** [comment]
Could you try `php artisan clear` after upgrading?

**Comment by jamesj2** [comment]
I tried that and verified the cached directory was empty. Also tried 'php optimize:clear'.

**Comment by jamesj2** [comment]
I did some more digging I found if I move the `$this->registerCustomBladeCompiler();` after `$this->registerBladeComponentsAndDirectives();` in `vendor/protonemedia/laravel-splade/src/ServiceProvider.php` and clear my view cache.  It fixes my issue.

**Comment by pascalbaljet** [comment]
I upgraded two apps and I can't find a way to break it. Could you share your `composer.json` file? Maybe there are packages that interact with the Blade Compiler/Engine.

**Comment by pascalbaljet** [comment]
Thanks for helping out! Fixed in v1.2.13, make sure to [update both packages](https://splade.dev/docs/upgrading) (JS + PHP).

**Comment by jamesj2** [comment]
You're too fast :)  I was trying to figure out which if any of my composer packages broke it.   I was able to break it with the packages below but since upgrading to 1.2.13 it working.  Thanks!

```
appstract/laravel-blade-directives
pharaonic/laravel-readable
php-flasher/flasher-toastr-laravel
pragmarx/yaml
```
